src/index.js

```diff
@@ -25,6 +25,25 @@ const server = http.createServer((req, res) => {
   // Check if health path
   if (req.url === readinessURL || req.url === livenessURL) {
     protect(req, res, () => res.end("OK"))
+    // Process Knative Events
+  } else if (('ce-type' in req.headers) && req.headers['ce-type'].startsWith('dev.knative')) {
+    var data = "";
+    req.setEncoding('utf8');
+
+    req.on('data', function (chunk) {
+      data += chunk;
+      if (data.length == parseInt(req.headers['content-length'], 10)) {
+        // Event 'end' is not emmited at the end of every request,
+        // we have to do that explicitly
+        req.emit('end');
+      }
+    });
+
+    req.on('end', function () {
+      req.body = data;
+      res.end(func(new Context(req, res)));
+    });
+    // Process standard HTTP requests
```

Comment:
Can you help me understand why you are modifying the incoming request for the `Context` object. Couldn't you just do all of this processing in the `Context` constructor?

Reply:
By modifying you mean setting `req.body = data;`? I thought that `body` is the right place where to store the incoming data and I didn't wanted to introduce another parameter in the `Context`. But I am open to any suggestions. Or do you mean anyting else?

Reply:
Yes, that's basically what I was questioning. It just seems odd to me to have the event processing broken into two separate files. The same `if` condition exists in both, so I wondered why you didn't just do all of the data stream processing in the `Context` constructor where you are unmarshalling it.

Reply:
You are right, this was leftover from previous design and I somehow forget about it. Thanks a lot, fixed.